src/provider/data.rs

```diff
@@ -484,7 +484,12 @@ static P_GMAIL: Lazy<Provider> = Lazy::new(|| {
         Server { protocol: Imap, socket: Ssl, hostname: "imap.gmail.com", port: 993, username_pattern: Email },
         Server { protocol: Smtp, socket: Ssl, hostname: "smtp.gmail.com", port: 465, username_pattern: Email },
     ],
-    config_defaults: None,
+    config_defaults: Some(vec![
+        ConfigDefault {
+            key: Config::DeleteToTrash,
```

Comment:
Done. But i introduced a new struct ProviderOptions for options like delete_to_trash that have reasonable defaults. Otherwise provider/data.rs will swell. Although it's auto-generated, its size now is O(nr_providers * nr_options) which isn't good (do not scale well)